### PR TITLE
Support block hash alias in console

### PR DESCRIPTION
**Pull request by promag**
Add support to use @best or @height instead of block hash in the
RPC console. The approach used is to simply replace those expressions
with the relevant RPC method.

### CODE REVIEW

**Comment by promag** [comment]
Close and reopen, travis didn't pick latest commit.

**Comment by promag** [comment]
Now using `QRegExp`, not sure why it fails to build with `QRegularExpression` since it's (supposedly) available since Qt 5.0.

**Comment by hebasto** [comment]
> Now using `QRegExp`, not sure why it fails to build with `QRegularExpression` since it's (supposedly) available since Qt 5.0.

https://github.com/bitcoin/bitcoin/blob/9366a73d695136fec8de12d3942254543903c0cd/depends/packages/qt.mk#L99:
```
$(package)_config_opts += -no-feature-regularexpression
```

**Comment by promag** [comment]
This is an alternative approach for the GUI change in https://github.com/bitcoin/bitcoin/pull/16439

File: src/qt/rpcconsole.cpp [hunk 1]
```diff
@@ -383,10 +384,13 @@ bool RPCConsole::RPCParseCommandLine(interfaces::Node* node, std::string &strRes
     }
 }
 
-void RPCExecutor::request(const QString &command, const WalletModel* wallet_model)
+void RPCExecutor::request(QString command, const WalletModel* wallet_model)
 {
     try
     {
+        command.replace(QRegExp("@best"), "getbestblockhash()");
+        command.replace(QRegExp("@([\\d]+)"), "getblockhash(\\1)");
```
Comment (ajtowns) [inline src/qt/rpcconsole.cpp:392]: Would it be better to have "\b@best\b" (and likewise for digits) so you don't replace "bob@bestdomainever.com" in a label or similar?

Comment (hebasto) [inline src/qt/rpcconsole.cpp:392]: The opening word boundary doesn't work with a "word" that starts with the `@` character. But the closing word boundary indeed improves behavior and error handling:
```suggestion
        command.replace(QRegExp("@best\\b"), "getbestblockhash()");
        command.replace(QRegExp("@([\\d]+)\\b"), "getblockhash(\\1)");
```

**Review by hebasto — CHANGES_REQUESTED** [review]
Approach ACK d2379e5c8f831abc244616069b6382626d6a6382.

Only `help-console` message provides a user with info about new aliases. The help messages of the affected RPC's still lack these updates, but I think it is OK for the GUI console.

This PR seems preferable than the [2f9f2d66cbada9134238f25aa12691876bdeff6b](https://github.com/bitcoin/bitcoin/pull/16439/commits/2f9f2d66cbada9134238f25aa12691876bdeff6b) commit from https://github.com/bitcoin/bitcoin/pull/16439.

File: src/qt/rpcconsole.cpp [hunk 2]
```diff
@@ -32,6 +32,7 @@
 #include <QKeyEvent>
 #include <QMenu>
 #include <QMessageBox>
+#include <QRegExp>
```
Comment (luke-jr) [inline src/qt/rpcconsole.cpp:35]: I don't think this warrants adding `QRegExp` to the project

File: src/qt/rpcconsole.cpp (hunk 1 above)
Comment (luke-jr) [inline src/qt/rpcconsole.cpp:392]: I think `RPCParseCommandLine` would be the proper place to do this.

**Review by luke-jr — CHANGES_REQUESTED** [review]

File: src/qt/rpcconsole.cpp (hunk 2 above)
Comment (promag) [reply src/qt/rpcconsole.cpp:35]: We already have it. Maybe you are thinking of `QRegularExpression`?

**Comment by hebasto** [comment]
@promag Any update on this? :)

**Comment by promag** [comment]
@hebasto I'm not convinced this is a good change. Closing for now and leaving it up for grabs.

**Pull request closed without merging**